rhp/v2/rhp.go

```diff
@@ -280,45 +303,116 @@ type (
 	}
 )
 
-// RPCSectorRootsCost returns the price of a SectorRoots RPC.
-func RPCSectorRootsCost(settings HostSettings, n uint64) types.Currency {
-	return settings.BaseRPCPrice.
-		Add(settings.DownloadBandwidthPrice.Mul64(n * 32)).  // roots
-		Add(settings.DownloadBandwidthPrice.Mul64(128 * 32)) // proof
+type RPCCost struct {
+	Base       types.Currency
+	Storage    types.Currency
+	Ingress    types.Currency
+	Egress     types.Currency
+	Collateral types.Currency
+}
+
+func (c RPCCost) Add(o RPCCost) RPCCost {
+	return RPCCost{
+		Base:       c.Base.Add(o.Base),
+		Storage:    c.Storage.Add(o.Storage),
+		Ingress:    c.Ingress.Add(o.Ingress),
+		Egress:     c.Egress.Add(o.Egress),
+		Collateral: c.Collateral.Add(o.Collateral),
+	}
+}
+
+func (c RPCCost) Total() (cost, collateral types.Currency) {
+	return c.Base.Add(c.Storage).Add(c.Ingress).Add(c.Egress), c.Collateral
 }
 
-// RPCReadCost returns the price of a Read RPC.
-func RPCReadCost(settings HostSettings, sections []RPCReadRequestSection) types.Currency {
-	sectorAccessPrice := settings.SectorAccessPrice.Mul64(uint64(len(sections)))
+// RPCReadCost returns the cost of a Read RPC.
+func RPCReadCost(sections []RPCReadRequestSection, proof bool, settings HostSettings) (RPCCost, error) {
```

Comment:
Should these be moved onto `HostSettings` as we did for the price table in RHP3?